src/packs/checker/package_todo.rs

```diff
@@ -0,0 +1,125 @@
+        let validation_errors: Vec<String> = configuration
+            .pack_set
+            .packs
+            .par_iter()
+            .filter_map(|pack| {
+                let package_todo_path =
+                    pack.yml.parent().unwrap().join("package_todo.yml");
+
+                // Skip packs that don't have package_todo.yml files
+                if !package_todo_path.exists() {
+                    return None;
+                }
+
+                validate_package_todo_format(
+                    &package_todo_path,
+                    &pack.name,
+                    configuration.packs_first_mode,
+                )
+                .err()
+            })
+            .collect();
+
+        if validation_errors.is_empty() {
+            None
+        } else {
+            Some(validation_errors)
+        }
+    }
+}
+
+/// Validates the format of a single package_todo.yml file.
+///
+/// This function implements the core validation logic:
+/// 1. Reads the current file content
+/// 2. Deserializes it to ensure it's valid YAML and matches PackageTodo structure
+/// 3. Re-serializes it using the standard serialization logic
+/// 4. Compares the result with the original content
+///
+/// # Arguments
+/// * `package_todo_path` - Path to the package_todo.yml file to validate
+/// * `pack_name` - Name of the pack (used for generating the correct header)
+/// * `packs_first_mode` - Whether the project uses packs.yml (affects command suggestions)
+///
+/// # Returns
+/// * `Ok(())` if the file is correctly formatted
+/// * `Err(String)` with a descriptive error message if validation fails
+///
+/// # Common causes of validation failures
+/// - Missing `---` separator after header comments
+/// - Incorrect ordering of violations or files (should be alphabetically sorted)
+/// - Manual edits that break the standard serialization format
+/// - Wrong header comment (should match packs_first_mode setting)
+fn validate_package_todo_format(
+    package_todo_path: &Path,
+    pack_name: &str,
+    packs_first_mode: bool,
+) -> Result<(), String> {
+    // Read the current file content
+    let current_content =
+        fs::read_to_string(package_todo_path).map_err(|e| {
+            format!("Failed to read {}: {}", package_todo_path.display(), e)
+        })?;
+
+    // Deserialize to ensure the file is valid and can be parsed
+    let package_todo: PackageTodo = serde_yaml::from_str(&current_content)
+        .map_err(|e| {
+            format!("Failed to parse {}: {}", package_todo_path.display(), e)
+        })?;
+
+    // Re-serialize using the standard serialization logic to get the expected format
+    let expected_content = crate::packs::package_todo::serialize_package_todo(
+        pack_name,
+        &package_todo,
+        packs_first_mode,
+    );
+
+    // Compare the current content with the expected serialized format
+    if current_content != expected_content {
+        return Err(format!(
```

Comment:
If using `anyhow`, this would need to be written as 

```suggestion
        return Err(anyhow!(
```

with a `use anyhow::anyhow` in the imports above.